app/src/test/java/ch/epfl/sdp/items/HealthpackTest.java

```diff
@@ -21,16 +23,16 @@ public void cloneHasSameHealthPoints() {
 
     @Test
     public void healthPackIncreasesHealthToThreshold(){
-        Player sick = new Player(20.0, 20.0, 100, "amroa", "amro.abdrabo@gmail.com");
-        sick.setHealthPoints(20.0);
-        PlayerManager.getInstance().setCurrentUser(sick);
-        Healthpack pack = new Healthpack(30);
-        //pack.useOn(sick);
-        //assertTrue(sick.getHealthPoints() == 50.0);
-        //Healthpack pack2 = new Healthpack(70);
-        //pack2.useOn(sick);
-        //assertTrue(sick.getHealthPoints() == 100.0);
-        // PlayerManager.getInstance().setCurrentUser(originalPlayer);
+        RandomGenerator randGen = new RandomGenerator();
+        Player sick = new Player(20.0, 20.0, 100, randGen.randomString(10), randGen.randomEmail());
+        sick.setHealthPoints(10.0);
+        RandomGenerator rand = new RandomGenerator();
+        Healthpack pack = rand.randomHealthPack();
+        pack.useOn(sick);
```

Comment:
Does this works fine? Because I think it has to go through the Server's synchronizing and thus, if the server did not execute it, I'm not sure it will work

Reply:
If you're talking about the useOn(player) method, then it works (i.e. the player's health will be updated), but no synchronization involved at this level (not needed for this test I think).

Reply:
Ok